src/System.Management.Automation/engine/remoting/server/serverremotesession.cs

```diff
@@ -206,7 +206,7 @@ internal class ServerRemoteSession : RemoteSession
             {
                 throw PSTraceSource.NewInvalidOperationException("RemotingErrorIdStrings.NonExistentInitialSessionStateProvider", configurationProviderId);
             }
-
+            senderInfo.ConfigurationName = configurationProviderId;
```

Comment:
It would be better to parse the configuration name from the shellUri (configurationProcviderId) here rather than parse it each time later when a transcript header is written.